solidity/contracts/converter/BancorConverter.sol

```diff
@@ -606,16 +624,17 @@ contract BancorConverter is IBancorConverter, SmartTokenController, ContractRegi
         uint256 reserveBalance = getReserveBalance(_reserveToken);
         assert(amount < reserveBalance || (amount == reserveBalance && _sellAmount == tokenSupply));
 
-        // destroy _sellAmount from the caller's balance in the smart token
-        token.destroy(msg.sender, _sellAmount);
+        // destroy _sellAmount from the converter balance in the smart token
+        token.destroy(this, _sellAmount);

```

Comment:
@yudilevi:
Why have you changed `msg.sender` to `this` here?

Reply:
Same as above